src/main/java/com/jorgepatrick/NumberConverter.java

```diff
@@ -72,25 +82,41 @@ public int parseRomanToArabic(final String romanNumber) {
 
         romanNumberValidator.validateRomanNumber(romanNumber);
 
-        int unit = 0;
+        int arabicNumber = 0;
 
         for (int currentDigit = 0; currentDigit < romanNumber.length(); currentDigit++) {
-            if (romanNumber.toUpperCase().charAt(currentDigit) == I.value().charAt(0)) {
-                int nextDigit = currentDigit + 1;
-                if (nextDigit == romanNumber.length() ||
-                    romanNumber.toUpperCase().charAt(nextDigit) == I.value().charAt(0)) {
-                    unit += 1;
-                } else {
-                    unit -= 1;
-                }
+
+            if (Arrays.stream(RomanOnes).toList().contains(romanNumber.substring(currentDigit, currentDigit + 1))) {
+                int digit = Arrays.stream(RomanOnes).toList().indexOf(romanNumber.substring(currentDigit, currentDigit + 1));
+                arabicNumber += getOnes(romanNumber, currentDigit, RomanDigits[digit]);
             }
 
             if (romanNumber.toUpperCase().charAt(currentDigit) == V.value().charAt(0)) {
-                unit += 5;
+                arabicNumber += 5;
             }
         }
+        return arabicNumber;
+    }
+
+    private int getOnes(String romanNumber, int currentDigit, Digits digit) {
+
+        int sum = 0;
+
+        int nextDigit = currentDigit + 1;
+        if (nextDigit == romanNumber.length() ||
+            isNextDigitEqualOrLess(romanNumber.toUpperCase().charAt(currentDigit), romanNumber.toUpperCase().charAt(nextDigit))) {
```

Comment:
Is it better to send the two chars as parameters (as is) or should I've sended the String and position?

Reply:
You can create a method that does this `.toUpperCase().charAt(currentDigit)`

Reply:
Does it make that much difference?
from:
romanNumber.toUpperCase().charAt(nextDigit))
to:
getUpperChar(romanNumber, currentDigit)

Reply:
What about romanNumber being another abstraction?

Reply:
Yes, it can be a class